topk-js/index.d.ts

```diff
@@ -0,0 +1,195 @@
+/* auto-generated by NAPI-RS */
+/* eslint-disable */
+export declare class Client {
+  constructor(config: ClientConfig)
+  collections(): CollectionsClient
+  collection(name: string): CollectionClient
+}
+
+export declare class CollectionClient {
+  get(id: string, fields?: Array<string> | undefined | null, lsn?: number | undefined | null, consistency?: ConsistencyLevel | undefined | null): Promise<Record<string, any>>
+  count(lsn?: number | undefined | null, consistency?: ConsistencyLevel | undefined | null): Promise<number>
+  query(query: query.Query, lsn?: number | undefined | null, consistency?: ConsistencyLevel | undefined | null): Promise<Array<Record<string, any>>>
+  upsert(docs: Array<Record<string, any>>): Promise<number>
+  delete(ids: Array<string>): Promise<number>
+}
+
+export declare class CollectionsClient {
+  list(): Promise<Array<Collection>>
+  create(name: string, schema: Record<string, schema.FieldSpec>): Promise<Collection>
+  get(name: string): Promise<Collection>
+  delete(name: string): Promise<void>
+}
+
+export interface ClientConfig {
+  apiKey: string
+  region: string
+  host?: string
+  https?: boolean
+  headers?: Record<string, string>
+}
+
+export interface Collection {
+  name: string
+  orgId: string
+  projectId: string
+  schema: Record<string, FieldSpec>
+  region: string
+}
+
+export type ConsistencyLevel =  'indexed'|
+'strong';
+
+export interface CreateCollectionOptions {
+  name: string
+  schema: Record<string, schema.FieldSpec>
+}
+
+export interface FieldSpec {
+  dataType: schema.DataType
+  required: boolean
+  index?: schema.FieldIndexUnion
+}
+
+export type FunctionExpression =
+  | { type: 'KeywordScore' }
+  | { type: 'VectorScore', field: string, query: data.Vector }
+  | { type: 'SemanticSimilarity', field: string, query: string }
+
+export interface Term {
+  token: string
+  field?: string
+  weight: number
+}
+
+export declare namespace data {
+  export class Vector {
+
+  }
+  export function binary(values: Array<number>): any
+  export function binaryVector(values: Array<number>): Vector
+  export function f32Vector(values: Array<number>): Vector
+  export function u8Vector(values: Array<number>): Vector
+  export type VectorUnion =
+    | { type: 'Float', values: Array<number> }
+    | { type: 'Byte', values: Array<number> }
+    | { type: 'Binary', values: Array<number> }
+}
+
+export declare namespace query {
+  export class LogicalExpression {
+    static create(expr: LogicalExpressionUnion): LogicalExpression
+    get expr(): LogicalExpressionUnion
```

Comment:
q: do we need `get expr()` to run equality checks only, or something else as well?

Reply:
just equality checks